source/components/composing-components.md

```diff
@@ -0,0 +1,260 @@
+{{/post-list}}
+```
+
+And will result in the following HTML:
+
+```html
+<div id="ember123" class="ember-view">
+  <h3>Tomster goes to town</h3>
+  <p>Greatest post ever!</p>
+  <h3>Tomster on vacation</h3>
+  <p>Greatest post ever!</p>
+</div>
+```
+
+But what use is it to just output the same thing over and over? Don't we want to customize our posts,
+and display the right content? Sure we do. Lets explore the `{{yield}}` helper a bit.
+
+### Data Down
+
+To accomplish composability beyond just simple templates, we need to pass context to those templates.
+This can be done with the `{{yield}}` helper that was just introduced above.
+
+The `{{yield}}` defines where the content we defined inside our component block will yield in the component's layout,
+as we saw in the previous section. Apart from that, the yield helper also allows us to send data down, providing a context for
+the content inside of the component's block. This allows anyone using our component to get context about what's going on
+inside, and to customize the output of the component with that information.
+
+```hbs
+{{yield}}
+{{yield "post"}}
+{{yield item}}
+{{yield this "footer"}}
+```
+
+By default yield does not send any context, but you can provide an arbitrary number of arguments.
+Sending data down allows the consumer of our component to utilize that data.
+For the consumer to get access to the data, she needs to know what data a component exposes,
+this is where documentation is so crucial, and from there the yielded data can be accessed
+with the `as` operator. Let's take the following template as an example:
+
+```app/components/user-list/template.hbs
+{{#each users as |user|}}
+  {{yield user}}
+{{/each}}
+```
+
+This `{{user-list}}` component will yield as many times as there are users.
+We can consume it in the following way:
+
+```app/users/template.hbs
+{{#user-list users=users as |user|}}
+  {{user-card user=user}}
+{{/user-list}}
+```
+
+Now `{{user-card}}` has access to the current user, which changes as the `{{user-list}}` iterates over each user
+since the `{{yield user}}` is inside the `{{each}}` block. Although this is nice, what if we used our component
+without a block, i.e. `{{user-list users=users}}`? This would make the component pretty useless since nothing is yielding,
+but the users are still being iterated. Let's mix in the `hasBlock` attribute and see if we can make it more useful.
+
+```app/components/user-list/template.hbs
+{{#if hasBlock}}
+  {{#each users as |user|}}
+    {{yield user}}
+  {{/each}}
+{{else}}
+  {{#each users as |user|}}
+    <section class="user-info">
+      <h3>{{user.name}}</h3>
+      <p>{{user.bio}}</p>
+    </section>
+  {{/each}}
+{{/if}}
+```
+
+Now we our component is a little more useful, because it has sane defaults, but it also allows us to override
+those defaults. I'm sure there is more that we can do to our `{{user-list}}` component to make it more useful,
+like handling different types of users.
```

Comment:
> I'm sure there is **more** that we can do to our `{{user-list}}` component to make it **more** useful

I'm not sure this sentence adds anything, it can likely be removed or replaces with a summary of what was done in this section if the goal is to summarize.